spec/requests/seance_spec.rb

```diff
@@ -0,0 +1,333 @@
+require 'rails_helper'
+
+RSpec.describe "/seances", type: :request do
+  let(:user) { create :user }
+  let(:manager) { create :user, email: "testmanager@test.com", role: :manager }
+
+  describe "GET /seances/new" do
+    context "when no user" do
+      it "redirects to sign_in" do
+        get("/seances/new")
+        expect(response).to redirect_to("/users/sign_in")
+      end
+
+      it "returns redirect status" do
+        get("/seances/new")
+        expect(response.status).to eq(302)
+      end
+    end
+
+    context "when user without permission" do
+      before { sign_in user }
+      it "redirects to root" do
+        get("/seances/new")
+        expect(response).to redirect_to("/")
+      end
+
+      it "returns redirect status" do
+        get("/seances/new")
+        expect(response.status).to eq(302)
+      end
+    end
+
+    context "when user with permission" do
+      before { sign_in manager }
+      it "returns successful response" do
+        get("/seances/new")
+        expect(response.status).to eq(200)
+      end
+
+      it "renders proper template" do
+        get("/seances/new")
+        expect(response).to render_template("seances/new")
+      end
+    end
+  end
+
+  describe "POST /seances" do
+    let(:params) do
+      {
+        seance: {
+          start_time:,
+          price:,
+          movie_id:,
+          hall_id:
+        }
+      }
+    end
+    let(:movie) { create :movie }
+    let(:hall) { create :hall }
+
+    let(:hall_id) { hall.id }
+    let(:movie_id) { movie.id }
+    let(:price) { Faker::Number.number(digits: 2) }
+    let(:start_time) { DateTime.current }
```

Comment:
FactoryBot has a method for generating param hashes
```suggestion
    let(:params) { attributes_for(:seance, start_time:, price:, movie:, hall:) }
    let(:movie) { create :movie }
    let(:hall) { create :hall }
    let(:price) { Faker::Number.number(digits: 2) }
    let(:start_time) { DateTime.current }
```